File: interface/src/app/core/services/entity.service.ts
import { Injectable } from '@angular/core';
import {ApiService} from './api.service';
import {Observable} from 'rxjs';
import {HttpParams} from '@angular/common/http';
import {EnoviaEntity} from '../../model/enovia-entity';
import {EntityGraph} from '../../model/entity-graph';

@Injectable()
export class EntityService {

  constructor(private apiService: ApiService) { }

  getEntities(releaseId: number, name: string, searchWord: string): Observable<EnoviaEntity[]>  {
    const params = new HttpParams().set('releaseId', releaseId.toString())
      .set('type', name)
      .set('searchWord', searchWord);

    return this.apiService.get('/entity', params);
  }
  getEntity(entityId: number): Observable<EnoviaEntity>  {
    return this.apiService.get('/entity/' + entityId);
  }
  saveDescription(entityId: number, entity: EnoviaEntity): Observable<EnoviaEntity>  {
    return this.apiService.put('/entity/' + entityId, entity);
  }

  getEntityGraph(entityId: number): Observable<EntityGraph> {
    return this.apiService.get('/entity/' + entityId + '/graph');
  }
}

File: src/main/java/ru/ase/ims/enomanager/service/EntityService.java
package ru.ase.ims.enomanager.service;

import ru.ase.ims.enomanager.model.EnoviaEntity;
import ru.ase.ims.enomanager.model.Release;
import ru.ase.ims.enomanager.model.graph.EntityGraph;

import javax.transaction.Transactional;
import java.util.List;

public interface EntityService {
    List<EnoviaEntity> getEntityList(Long releaseId, String type, String searchWord);
    List<EnoviaEntity> findEntities(Release release);
    EnoviaEntity updateEntity(Long entityId, EnoviaEntity enoviaEntity);
    @Transactional
    EntityGraph getEntityGraph(Long entityId);
    EnoviaEntity getEntity(Long entityId);
}
